src/column.rs

```diff
@@ -514,13 +514,30 @@ impl Column {
 		Ok(())
 	}
 
-	pub fn write_stats(&self, file: &std::fs::File) {
+	pub fn write_stats(&self, writer: &mut impl std::io::Write) {
 		let tables = self.tables.read();
 		tables.index.write_stats(&self.stats);
-		self.stats.write_summary(file, tables.index.id.col());
+		self.stats.write_summary(writer, tables.index.id.col());
 	}
 
-	pub fn iter_while(&self, log: &Log, mut f: impl FnMut (u64, Key, u32, Vec<u8>) -> bool) -> Result<()> {
+	pub fn clear_stats(&self) {
+		let tables = self.tables.read();
+		let empty_stats = ColumnStats::empty();
+		tables.index.write_stats(&empty_stats);
+	}
+
+	pub fn iter_while(&self, log: &Log, f: impl FnMut (u64, Key, u32, Vec<u8>) -> bool) -> Result<()> {
+		self.iter_while_inner(log, f, 0,  None, |_, _| false)
+	}
+
+	fn iter_while_inner(
```

Comment:
Feels like this function is now doing too much. Maybe let's keep a separate one for checking consistency. Also you don't need start and end chunk when you have the callback. 
So it should either be  `fn check_consistency(&self, log: &Log, start_chunk: u64, sub_index: u8, end_chunk: Option<u64>)-> Result<(), ConsistencyError>` which returns on the first error. `ConsistencyError` contains chunk, entry with sub index so that you can continue checking from the next entry.

Or it could be `check_consistency(&self, log: &Log, f: impl FnMut (Result<Progress, ConsistencyError>) -> bool`
which calls `f` for each with progress or error information and decides if to stop based on what the callback returns.

Reply:
Also, consistency check should not use preimage iteration over the value table, as the index is not really validated in this mode.

Reply:
Yes 'end_chunk' does not make sense.
I did plan to suspend/restore iteration on error at first, but it would need to manage sub_index when starting, using a call back was a bit easier. So will use an enum parameter on the callback instead.